zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/PendingProcessMessageSubscriptionChecker.java

```diff
@@ -67,7 +67,12 @@ public void onResumed() {
 
   private void rescheduleTimer() {
     if (schouldRescheduleTimer) {
-      scheduleService.runDelayed(SUBSCRIPTION_CHECK_INTERVAL, this::checkPendingSubscriptions);
+      scheduleService.runAt(
+          ActorClock.currentTimeMillis() + SUBSCRIPTION_CHECK_INTERVAL.toMillis(),
+          taskResultBuilder -> {
+            checkPendingSubscriptions();
+            return taskResultBuilder.build();
```

Comment:
I'm curious if adding `taskResultBuilder.build()` here introduces any change to the original functionality? was that intentional?

Reply:
That's a good point. The task is transformed into a runnable that something special: attempt to write the result batch to the log stream. What if the task result builder is empty? 

It may be better still to take the other approach of overloading the `runAt` method for a runnable. We avoided this originally, because we thought it wouldn't make a difference. Let's verify that it really doesn't make a difference.

Reply:
:+1: Will look into this tomorrow